### PR TITLE
refactor(node): Use r.js instead of the home grown amd loader. 

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Use the supported r.js to load the project in node.

@vladikoff - this builds on your PR and fixes the issue with loading es6-promise in node. 

If you think it's good to go, let's close #75 and just merge this!

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -21,8 +21,9 @@
     "tests"
   ],
   "dependencies": {
-    "p-promise": "https://github.com/rkatic/p.git#3c1fe61c9ba47aa3c09a55eb3025bd29a238d1ea",
-    "micrajax": "0.0.5"
+    "p-promise": "0.5.0",
```
Comment (vladikoff) [inline bower.json]: todo: remove

**Comment by vladikoff** [comment]
> 604 additions, 2,857 deletions not shown 

did shrinkwrap lose a lot of files?

**Comment by shane-tomlinson** [comment]
> did shrinkwrap lose a lot of files? 

I did an `npm shrinkwrap` using Ubuntu, node v0.10.40, and npm 2.4.13. I did not add the `--production` flag.

**Comment by vladikoff** [comment]
> I did not add the --production flag.

maybe we had dev stuff added to shrinkwrap, so you need to add `--dev` because it would only wrap `production` by default. Or maybe shrinkwrap files for slimmer

**Comment by shane-tomlinson** [comment]
Closing, this won't work suit webpack and browserify.

**Pull request closed without merging**